src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveLevelService.kt

```diff
@@ -67,6 +71,10 @@ class IncentiveLevelService(
     return incentiveLevelRepository.save(incentiveLevel)
       .toDTO()
       .also {
+        if (it.required) {
+          // This probably could be moved out of the transaction once auditing is moved out:
```

Comment:
Assuming enabling the level in all prisons is quick enough it may make sense to keep this within the same transaction.

The two operations kinda make sense only together no? 🤔

Reply:
well i'm pretty sure it would be up to 120 _separate_ DB updates cos the values come from this service and the `repository.saveAll(…)` returns a flow – that implies it updates one-by-one. i don't mind the prisons picking up the change "a little later". still, right now it's inside the transaction :)